### PR TITLE
fix(core): cross-browser viewport width for mobile

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/cdk/utils/dom/get-viewport-width.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export function getViewportWidthBy(windowRef: Window): number {
```
Comment (nsbarsukov) [inline projects/cdk/utils/dom/get-viewport-width.ts:1]: `tui`-prefix ?

File: projects/cdk/utils/dom/get-viewport-height.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export function getViewportHeightBy(windowRef: Window): number {
```
Comment (nsbarsukov) [inline projects/cdk/utils/dom/get-viewport-height.ts:1]: `tui`-prefix ?

File: projects/core/tokens/is-mobile-resolution.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ export const TUI_IS_MOBILE_RES = new InjectionToken<Observable<boolean>>(
             return typedFromEvent(windowRef, 'resize').pipe(
                 share(),
                 startWith(null),
-                map(() => windowRef.innerWidth < mobile),
+                map(() => getViewportWidthBy(windowRef) < mobile),
```
Comment (nsbarsukov) [inline projects/core/tokens/is-mobile-resolution.ts:19]: Should we use this util in other places ?
Screenshot 2022-06-22 at 11 15 59

**Review by nsbarsukov — APPROVED** [review]

File: projects/core/tokens/is-mobile-resolution.ts (hunk 3 above)
Comment (splincode) [reply projects/core/tokens/is-mobile-resolution.ts:19]: @waterplea what do you think?

File: projects/cdk/utils/dom/get-viewport-height.ts (hunk 2 above)
Comment (waterplea) [reply projects/cdk/utils/dom/get-viewport-height.ts:1]: Also `By` is unnecessary and let's destruct :)

File: projects/core/tokens/is-mobile-resolution.ts (hunk 3 above)
Comment (waterplea) [reply projects/core/tokens/is-mobile-resolution.ts:19]: Yes, makes sense to use it everywhere

**Review by waterplea — CHANGES_REQUESTED** [review]

File: projects/cdk/utils/dom/get-viewport-height.ts (hunk 2 above)
Comment (splincode) [reply projects/cdk/utils/dom/get-viewport-height.ts]: fixed

File: projects/core/tokens/is-mobile-resolution.ts (hunk 3 above)
Comment (splincode) [reply projects/core/tokens/is-mobile-resolution.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**